components/public-api/typescript-common/src/user-utils.ts

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { User } from "@gitpod/public-api/lib/gitpod/v1/user_pb";
+import { User as UserProtocol } from "@gitpod/gitpod-protocol/lib/protocol";
+
+/**
+ * Returns a primary email address of a user.
+ *
+ * For accounts owned by an organization, it returns the email of the most recently used SSO identity.
+ *
+ * For personal accounts, first it looks for a email stored by the user, and falls back to any of the Git provider identities.
+ *
+ * @param user
+ * @returns A primaryEmail, or undefined.
+ */
+export function getPrimaryEmail(user: User): string | undefined {
```

Comment:
is it copy of one from protocol? How we are going to keep them in sync?

Would it make sense to translate to protocol, perform old code and then convert back?

Reply:
Good question. Ideally, we remove both copies. We've an issue to persist the email addresses in `User.emails`, which would result in simple read, no computation.

Reply:
w.r.t syncing, I think it would be possible to use a single for both User shapes.

Reply:
Thanks for pushing on this! I just made sure this utterly ugly thing isn't duplicated. It's now _moved_ from protocol to common package. Same applies to `getProfile`, which actually shouldn't exist at all.